### DISCUSSION TITLE
Support for standalone derivative datasets

### DISCUSSION THREAD

**Discussion by ghisvail** [ANCPLabOldenburg/ancp-bids#68] [Q&A]
I was wondering whether `ancpbids` supported construction of [standalone derivative datasets](https://bids-specification.readthedocs.io/en/stable/02-common-principles.html#storage-of-derived-datasets) (2nd case)?

Processing tools may favour keeping source datasets read-only when running pipelines to prevent accidental alteration. So far, I was only able to create a derivative dataset from a source one using `ancpbids`. If that's not possible today, I believe it would be a significant improvement to the library.

**Comment by erdalkaraca** [comment]
Sounds doable :-)

Prerequisite:
- support writing back datasets that have been constructed in in-memory via the API
- allow to control whether writing back a derivative or a dataset

**Comment by ghisvail** [comment]
> Sounds doable :-)
>
Great!

> Prerequisite:
>
>    - support writing back datasets that have been constructed in
>    in-memory via the API
>    - allow to control whether writing back a derivative or a dataset
>
> What do you reckon would be the time required to implement this feature?

I'd be happy to implement it provided some mentorship on the internals of
the code.

>
>    -
>
>

**Reply by erdalkaraca** [reply]
> I'd be happy to implement it provided some mentorship on the internals of the code.

That would be great, I can offer a pair programming session to get you started.

**Reply by ghisvail** [reply]
Brilliant, I'll contact you by MP to define a schedule.
